app/src/main/kotlin/app/youkai/ui/feature/settings/view/PreferenceView.kt

```diff
@@ -0,0 +1,138 @@
+
+    private var checkableType: Int = CHECKABLE_TYPE_NONE
+    private var initialDescriptionText: CharSequence? = null
+    private var initialDisabledDescriptionText: CharSequence? = null
+    private var initialTitleText: CharSequence? = null
+    private var startEnabled: Boolean = true
+
+    private var checkableView: CompoundButton? = null
+    private var descriptionView: TextView? = null
+    private var titleView: TextView? = null
+
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        parseAttributes(attrs)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?, @AttrRes defStyleAttr: Int) :
+            super(context, attrs, defStyleAttr) {
+        parseAttributes(attrs)
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    constructor(context: Context, attrs: AttributeSet?, @AttrRes defStyleAttr: Int,
+                @StyleRes defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
+        parseAttributes(attrs)
+    }
+
+    var description: CharSequence?
+        get() = descriptionView?.text
+        set(description) { descriptionView?.text = description }
+
+    override fun dispatchRestoreInstanceState(container: SparseArray<Parcelable>) {
+        dispatchThawSelfOnly(container)
+    }
+
+    override fun dispatchSaveInstanceState(container: SparseArray<Parcelable>) {
+        dispatchFreezeSelfOnly(container)
+    }
+
+    override fun onFinishInflate() {
+        super.onFinishInflate()
+
+        val layoutResId = when (checkableType) {
+            CHECKABLE_TYPE_NONE -> {
+                if (initialDescriptionText.isNullOrBlank()) {
+                    R.layout.view_preference
+                } else {
+                    R.layout.view_preference_with_description
+                }
+            }
+            CHECKABLE_TYPE_CHECK_BOX -> {
+                if (initialDescriptionText.isNullOrBlank()) {
+                    R.layout.view_checkbox_preference
+                } else {
+                    R.layout.view_checkbox_preference_with_description
+                }
+            }
+            CHECKABLE_TYPE_SWITCH_COMPAT -> {
+                if (initialDescriptionText.isNullOrBlank()) {
+                    R.layout.view_switchcompat_preference
+                } else {
+                    R.layout.view_switchcompat_preference_with_description
+                }
+            }
+            else -> {
+                throw RuntimeException("illegal checkableType ($checkableType)")
+            }
+        }
+
+        LayoutInflater.from(context).inflate(layoutResId, this)
+        checkableView = findViewById(R.id.checkableView)
+        descriptionView = findViewById(R.id.descriptionView)
+        titleView = findViewById(R.id.titleView)
+
+        title = initialTitleText
+        description = if (initialDisabledDescriptionText.isNullOrBlank())
+            initialDescriptionText else initialDisabledDescriptionText
```

Comment:
I think we can safely say that as a convention we are always using curly braces with `if` line breaks.

Reply:
✔️